File: Arrays/Longest-Subarray-with-Sum-K/JavaScript/Longest-Subarray-with-Sum-K-Brute Force.js
/**
 * Problem: Longest Subarray with Sum K
 * Approach: Brute Force
 *
 * Generate every subarray and maintain its sum.
 *
 * Time Complexity: O(n^2)
 * Space Complexity: O(1)
 */

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestSubarray = function(nums, k) {
    let longest = 0;

    for (let i = 0; i < nums.length; i++) {
        let currentSum = 0;

        for (let j = i; j < nums.length; j++) {
            currentSum += nums[j];

            if (currentSum === k) {
                longest = Math.max(longest, j - i + 1);
            }
        }
    }

    return longest;
};

File: Arrays/Longest-Subarray-with-Sum-K/Python/Longest-Subarray-with-Sum-K-Brute Force.py
"""
Problem: Longest Subarray with Sum K
Approach: Brute Force

Generate every subarray and maintain its sum.
Whenever the sum equals k, update the maximum length.

Time Complexity: O(n^2)
Space Complexity: O(1)
"""

class Solution:
    def longestSubarray(self, nums: list[int], k: int) -> int:
        longest = 0

        for i in range(len(nums)):
            current_sum = 0

            for j in range(i, len(nums)):
                current_sum += nums[j]

                if current_sum == k:
                    longest = max(longest, j - i + 1)

        return longest